Index: CMakeLists.txt
===========================================================
--- CMakeLists.txt 2013-03-20 20:31:13.464811226 +0100
+++ CMakeLists.txt 2013-03-20 21:13:57.983755171 +0100
@@ -8,13 +8,13 @@
 set(Boost_USE_STATIC_LIBS ON)
 set(Boost_USE_MULTI_THREADED ON)
 find_package( Boost 1.45.0 REQUIRED unit_test_framework system regex date_time thread filesystem program_options )
-find_package( OpenSSL )
-find_package( Threads )
-set(CMAKE_VERBOSE_MAKEFILE true)
 
-if (CMAKE_BUILD_TYPE MATCHES Debug)
-    add_definitions(-DBOOST_NETWORK_DEBUG)
+option( WITH_SSL "enables HTTPS. requires OpenSSL" OFF )
+
+if (WITH_COMPILER)
+    find_package( OpenSSL )
 endif()
+find_package( Threads )
 
 if (OPENSSL_FOUND)
     add_definitions(-DBOOST_NETWORK_ENABLE_HTTPS)
@@ -28,13 +28,6 @@
       add_definitions(-D_WIN32_WINNT=0x0501)
     endif(WIN32)
     include_directories(${Boost_INCLUDE_DIRS})
-    enable_testing()
     add_subdirectory(libs/network/src)
-    add_subdirectory(libs/network/test)
-    if (NOT MSVC)
-      add_subdirectory(libs/mime/test)
-    endif(NOT MSVC)
-    add_subdirectory(libs/network/example)
 endif(Boost_FOUND)
 
-enable_testing()

Index: libs/network/src/CMakeLists.txt
===========================================================
--- libs/network/src/CMakeLists.txt 2013-03-20 21:17:57.973749926 +0100
+++ libs/network/src/CMakeLists.txt 2013-03-20 21:21:52.018744810 +0100
@@ -16,3 +16,19 @@
 
 set(CPP-NETLIB_HTTP_CLIENT_SRCS client.cpp)
 add_library(cppnetlib-client-connections ${CPP-NETLIB_HTTP_CLIENT_SRCS})
+
+install(
+    TARGETS
+        cppnetlib-uri
+        cppnetlib-server-parsers
+        cppnetlib-client-connections
+    DESTINATION
+        lib
+)
+install(
+    DIRECTORY
+        "${CMAKE_SOURCE_DIR}/boost"
+    DESTINATION
+        include
+)
+
