--- bofh_main.c	2013-05-01 15:28:15.301450965 +0200
+++ bofh_main.c	2013-05-01 15:29:23.514449474 +0200
@@ -78,8 +78,8 @@
 
 	bofh_proc_file->read_proc	= &bofh_content;
 	bofh_proc_file->mode		= S_IFREG | S_IRUGO;
-	bofh_proc_file->uid		= 0;
-	bofh_proc_file->gid		= 0;
+	bofh_proc_file->uid.val		= 0;
+	bofh_proc_file->gid.val		= 0;
 	bofh_proc_file->size		= 0;
 
 	pr_debug("BOFH module inititialized!\n");
